Mathlib/CategoryTheory/Localization/CalculusOfFractions.lean

```diff
@@ -0,0 +1,205 @@
+/-
+Copyright (c) 2023 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.Localization.Opposite
+
+/-!
+# Calculus of fractions
+
+Following the definitions by [Gabriel and Zisman][gabriel-zisman-1967],
+given a morphism property `W : MorphismProperty C` on a category `C`,
+we introduce the class `W.HasLeftCalculusOfFractions`. The main
+result (TODO) is that if `L : C ⥤ D` is a localization functor for `W`,
+then for any morphism `L.obj X ⟶ L.obj Y` in `D`, there exists an auxiliary
+object `Y' : C` and morphisms `g : X ⟶ Y'` and `s : Y ⟶ Y'`, with `W s`, such
+that the given morphism is a sort of fraction `g / s`, or more precisely of
+the form `L.map g ≫ (Localization.isoOfHom L W s hs).inv`.
+
+## References
+
+* [P. Gabriel, M. Zisman, *Calculus of fractions and homotopy theory*][gabriel-zisman-1967]
+
+-/
+
+namespace CategoryTheory
+
+open Category
+
+namespace Functor
+
+lemma congr_map_conjugate {C D : Type _} [Category C] [Category D] {F₁ F₂ : C ⥤ D}
```

Comment:
I guess this lemma could sensibly move to some basic file on functors?

Reply:
Thanks! Indeed, this lemma shall be used only in later PRs. I have removed it for now.